### PR TITLE
snakemake: 6.0.5 -> 6.5.0

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
[6.5.0] - 2021-06-22
====================
- Allow to set the default profile via the environment variable $SNAKEMAKE_PROFILE.
- There is a new default resource tmpdir (by default reflects the system setting), which is automatically used for temporary files by shell commands and scripts which properly consider the usual environment variables like $TMP, $TEMP, $TMPDIR (johanneskoester).
- The CLI flags --jobs and --cores are now separated, with --cores being responsible for local cores and global cores in the cluster case, and --jobs being responsible for number of jobs. Still -j and --jobs works as a fallback for local execution (johanneskoester).
- Added the ability to overwrite resources via --set-resources (johanneskoester).
- Various fixes for Windows execution (melund).
- Fixed a bug with fractional resources (johanneskoester).
- Fixed timeouts and other issues in google life science backend (johanneskoester).
- Fixed a bug with missing conda frontend definitions in subworkflows (johanneskoester).
- Skip envvar checking during linting (johanneskoester).
- Fixed a bug causing container images in modules to be ignored (johanneskoester).


[6.4.1] - 2021-05-27
====================
- Fixed bug in ``workflow.source_path()`` that occurred with modules included from remote locations (johanneskoester).
- Inform cluster jobs about conda/mamba/activate path such that they don't need to determine this themselves (johanneskoester).

[6.4.0] - 2021-05-20
====================
- Improvements in the docs (resource usage, best practices, remote files) (johanneskoester, admorris).
- functions given to params can now safely open input files generated by previous rules. If they are not present, TBD will be displayed and function will be reevaluated immediately before the job is executed (i.e. when files are present) (ASLeonard).
- Connection pool for SFTP and FTP remote files, increasing download performance (jmeppley).
- Require correct minimum version of smart_open (Redmar-van-den-Berg).
- Added workflow.source_path(path), allowing to get the correct path relative to the current Snakefile, even when Snakefile is included via URL (johanneskoester).
- Fixed bugs in module system (johanneskoester, dlaehnemann).
- Fixed issue with checkpoints and ruleorder where phantom dependencies are not properly removed from the DAG (jmeppley, johanneskoester).
- Disable tibanna behavior that opens a browser window for each job (nigiord).
- Allow ``Paramspace(..., filename_params="*")``, meaning that all columns of the paramspace will be encoded into the filename (kpj).
- Avoid PATH modification in cluster jobs (johanneskoester).
- For large sets of input files, pass files to wait for (FS latency) as a file instead of command line args (kpj, epruesse).

[6.3.0] - 2021-04-29
====================
- Changed behavior of ``workflow.snakefile`` to always point to the current file instead of the main Snakefile (also in case of includes and modules) (johanneskoester).
- Fixed a typo in an error message (nikostr).

[6.2.0] - 2021-04-22
====================
- Support for integration of foreign workflow management systems by introducing a ``handover`` directive that passes on all resources to a particular rule (which can then invoke another workflow management system). See the docs ("Integrating foreign workflow management systems") (johanneskoester).
- Behavior improvement for temp handling of checkpoint rules (epruesse).
- Several improvements in the docs (johanneskoester).

[6.2.1] - 2021-04-20
====================
- Fixed a minor bug in the linter.

[6.2.0] - 2021-04-20
====================
- Fixed several glitches in paramspace implementation (handling of bools, returning scalar values) (kpj).
- Fixed bugs in module implementation (dlaehnemann, johanneskoester).
- Fall back to greedy scheduling solver if ILP solver needs more than 10 sec (johanneskoester).

[6.1.1] - 2021-04-07
====================
- Fixed several small bugs of the new module system (johanneskoester, dlaehnemann).
- Fixed archive based conda deployment (johanneskoester).
- Better handling of download and target attributed in the interactive report (johanneskoester).

[6.1.0] - 2021-04-01
====================
- Snakemake now uses **mamba** as the default conda frontend (which can be overwritten by specifying to use conda via the --conda-frontend flag) (johanneskoester).
- Profiles using --cluster option can now handle relative submit script paths in combination with arguments (kdm9).
- New AutoRemoteProvider, which infers the type of remote file protocol from the given URL (kpj).
- When using global container directive, container usage can be deactivated on a per rule base (bilke).
- Bugfixes for checkpoint handling (johanneskoester).
- Bugfixes for the module system (johanneskoester, dlaehnemann).
- Various improvements for the tutorial.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127829` at 73725913 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>python38Packages.connection_pool</li>
    <li>python38Packages.stopit</li>
    <li>python39Packages.connection_pool</li>
    <li>python39Packages.stopit</li>
    <li>snakemake</li>
  </ul>
</details>

**Review by renatoGarcia — APPROVED** [review]

File: pkgs/development/python-modules/connection_pool/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "connection_pool";
+  version = "0.0.3";
+  disabled = !isPy3k;
+
+  # tests are missing from the PyPi tarball
+  src = fetchPypi {
+    inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/connection_pool/default.nix:14]:
```suggestion
  pname = "connection-pool";
  version = "0.0.3";

  # tests are missing from the PyPi tarball
  src = fetchPypi {
    pname = "connection_pool";
    inherit pname version;
```

File: pkgs/development/python-modules/stopit/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  version = "1.1.2";
+
+  # tests are missing from the PyPi tarball
+  src = fetchFromGitHub {
+    owner = "glenfant";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-uXJUA70JOGWT2NmS6S7fPrTWAJZ0mZ/hICahIUzjfbw=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/stopit/default.nix:17]:
```suggestion

  # no tests
  doCheck = false;

```

File: pkgs/development/python-modules/stopit/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+}:
+
+buildPythonPackage rec {
+  pname = "stopit";
+  version = "1.1.2";
+
+  # tests are missing from the PyPi tarball
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/stopit/default.nix:10]:
```suggestion
```

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -1337,6 +1337,8 @@ in {
 
   connect-box = callPackage ../development/python-modules/connect_box { };
 
+  connection_pool = callPackage ../development/python-modules/connection_pool { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1340]:
```suggestion
  connection-pool = callPackage ../development/python-modules/connection-pool { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/connection_pool/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/python-modules/connection_pool/default.nix]: done

File: pkgs/development/python-modules/stopit/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/development/python-modules/stopit/default.nix:17]: This is wrong, there are tests

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (veprbl) [reply pkgs/top-level/python-packages.nix]: done

File: pkgs/development/python-modules/stopit/default.nix (hunk 3 above)
Comment (veprbl) [reply pkgs/development/python-modules/stopit/default.nix:10]: no

File: pkgs/development/python-modules/connection-pool/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+, fetchPypi
+, isPy3k
+}:
+
+buildPythonPackage rec {
+  pname = "connection-pool";
+  version = "0.0.3";
+  disabled = !isPy3k;
+
+  # tests are missing from the PyPi tarball
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/connection-pool/default.nix:12]: Why not fetch from Github then?

File: pkgs/development/python-modules/connection-pool/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+}:
+
+buildPythonPackage rec {
+  pname = "connection-pool";
+  version = "0.0.3";
+  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/connection-pool/default.nix:10]:
```suggestion
```
No longer required after python2 split.

File: pkgs/development/python-modules/connection-pool/default.nix (hunk 5 above)
Comment (veprbl) [reply pkgs/development/python-modules/connection-pool/default.nix:12]: This comment is the unwanted one. Seems like it just got copied over from stopit.

**Pull request merged**